resources/assets/react/screens/Profile/AssociativeCareer.js

```diff
@@ -0,0 +1,119 @@
+/**
+ * Display all associations of a given user.
+ *
+ * @author Corentin Mercier <corentin@cmercier.fr>
+ *
+ * @copyright Copyright (c) 2019, SiMDE-UTC
+ * @license GNU GPL-3.0
+ */
+
+import React from 'react';
+import { connect } from 'react-redux';
+import { NavLink } from 'react-router-dom';
+import AssoCard from '../../components/AssoCard';
+
+import actions from '../../redux/actions';
+
+@connect(store => ({
+	config: store.config,
+	semesters: store.getData(['semesters']),
+	semestersFetched: store.isFetched(['semesters']),
+	roles: store.getData('roles'),
+	rolesIsFetching: store.isFetching('roles'),
+	rolesFetched: store.isFetched('roles'),
+}))
+class AssociativeCareerScreen extends React.Component {
+	constructor(props) {
+		super(props);
+
+		this.state = {
+			semesters: [],
+		};
+	}
+
+	componentDidMount() {
+		const {
+			user: { name },
+			dispatch,
+			rolesFetched,
+			semesters,
+		} = this.props;
+
+		if (!rolesFetched) {
+			dispatch(actions.roles.all());
+		}
+
+		dispatch(actions.config({ title: `${name} - Mon Parcours` }));
+
+		const newSemesters = [];
+		semesters
+			.slice()
+			.reverse()
+			.forEach(semester => {
+				actions.user.assos.all({ semester: semester.id }).payload.then(({ data }) => {
+					semester.assos = data;
+					newSemesters.push(semester);
```

Comment:
You should not push in an array but more in an object.
Here, you think that everything is resolved in order, but if a semester is returned faster, you'll get them in desorder.

Set your state like this: semesters[semester_id] = data

And after, in order to print, create a loop for each semester ;)